### PR TITLE
feat(onboarding): reusable workflow for automated onboarding

### PR DESCRIPTION
**Pull request by jiparis**
This a reusable workflow that uses the latest improvements on API_TOKENs to automatically onboard a GH job in Chainloop.
It needs a `project` parameter, and accepts a `workflow_name` as input, and, if missing, it will use the github action file name as the workflow name. 

Usage:
file .github/workflows/my_workflow.yml
```
jobs:
  onboard:
    uses: ./.github/workflows/chainloop_onboard.yml
    with:
      project: my-team
    secrets:
      api_token: ${{ secrets.CHAINLOOP_API_TOKEN }}
```

As a result, it will create a Chainloop workflow called `my_workflow` with the default contract.

### CODE REVIEW

**Review by migmartri — APPROVED** [review]
Cool!

File: .github/workflows/chainloop_onboard.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+on:
+  workflow_call:
+    inputs:
+      project:
+        description: The chainloop project, usually matching a project or team within the organization
```
[the remark this answers is not in the document]
Comment (javirln) [reply .github/workflows/chainloop_onboard.yml]: I've removed the hard requirement of the project and if not passed it's computed based on the `github.repositoryUrl`

[the remark this answers is not in the document]
Comment (migmartri) [reply .github/workflows/chainloop_onboard.yml]: I like it. 

but I'll leave it to @jiparis to approve this PR :)

Thanks!

File: .github/workflows/chainloop_onboard.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  chainloop_onboard:
+    name: Automatic Chainloop onboarding flow from Github Actions
+    runs-on: ubuntu-latest
+    outputs:
+      workflow_name: ${{ steps.set_workflow_name.outputs.workflow_name }}
+
+    steps:
+      - name: Install Chainloop
+        run: |
+          curl -sfL https://docs.chainloop.dev/install.sh | bash -s
```
[the remark this answers is not in the document]
Comment (migmartri) [reply .github/workflows/chainloop_onboard.yml:29]: Nevermind, these are different jobs, I guess that we'll need to start using Github cache for that cache

**Pull request merged**